modules/data.remote/inst/appeears2pecan.py

```diff
@@ -0,0 +1,206 @@
+    
+    end (str) -- ending date area of the data request in the form YYYY-MM-DD
+
+    product (str) -- product name followed by " . " and the product version, e.g. "SPL3SMP_E.003"
+
+    projection (str) -- type of projection, only required for polygon AOI type. None by default
+
+    Returns
+    -------
+    Nothing:
+            Output files are saved in the specified directory.
+            Output file is of netCDF type when AOI is a Polygon and csv type when AOI is a Point.
+    """
+
+    # API url
+    api = "https://lpdaacsvc.cr.usgs.gov/appeears/api/"
+
+    def authenticate():
+        """
+        uses the user provided NASA Earthdata credentials to request a token.
+        
+        Returns
+        -------
+        head (dict) : header contatining the authentication token 
+        """
+        try:
+            # if the user does not want to enter their credentials everytime they use this function, they need to store their username and password in a JSON file, preferabbly not in a git initialized directory
+            with open("path/to/cred.json", "r") as f:
+                cred = json.load(f)
+                user = cred["username"]
+                password = cred["password"]
+        except:
+            # if user does not want to store the credentials
+            user = getpass.getpass(prompt="Enter NASA Earthdata Login Username: ")
+            password = getpass.getpass(prompt="Enter NASA Earthdata Login Password: ")
+        # use the credentials to call log in service. Use request's HTTP Basic Auth to do the authentication
+        response = r.post("{}login".format(api), auth=(user, password))
+        # delete the user and password variables as they are no longer needed
+        del user, password
+        # raise an exception if the POST request returned an unsuccessful status code
+        response.raise_for_status()
+        token_response = response.json()
+        # extract the token
+        token = token_response["token"]
+        head = {"Authorization": "Bearer {}".format(token)}
+        return head
+
+    head = authenticate()
+
+    # query the available layers for the product and store it in a layer + product tuple
+    lst_response = r.get("{}product/{}".format(api, product)).json()
+    l = list(lst_response.keys())
+    layers = [(product, each_l) for each_l in l]
+    prodLayer = [({"layer": l[1], "product": l[0]}) for l in layers]
+
+    # special case to handle SMAP products
+    # SMAP products individually have more than 40 layers all of which are not allowed by the API to be downloaded in a single request
+    # if the requested product is one of the SMAP products then select the first 25 layers
+    if product in [
+        "SPL3FTP.002",
+        "SPL3SMP.006",
+        "SPL3SMP_E.003",
+        "SPL4CMDL.004",
+        "SPL4SMGP.004",
+    ]:
+        # change this part to select your own SMAP layers
+        prodLayer = prodLayer[0:25]
```

Comment:
this bit needs to throw a warning at least that subsetting happens

Reply:
Not related to this specific instance, once a question which I have is would the print statements, warnings etc from python be visible when executing the pecan workflow? Because I can't use the pecan logger functions.

Reply:
Good question, I don't know really, does reticulate pass along logs etc?

Reply:
as far as I know, print statements can be passed but not logs. This will make it very difficult to debug when something goes wrong.